sources/Algo/src/RPCLDefinitionAlgorithm.cpp

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2023, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+#include "Constants.h"
+#include "RPCLDefinitionAlgorithm.h"
+
+namespace dfl {
+namespace algo {
+
+void
+RPCLDefinitionAlgorithm::removeRpclInGeneratorsIfMissingConnexionToSVC() {
+    const std::map<std::string, inputs::AssemblingDataBase::DynamicAutomaton>& automatonsConfig = assembling_.dynamicAutomatons();
+    for (const auto &automaton : automatonsConfig) {
+        if (models_.count(automaton.second.id) == 0) {
+            continue;
+        }
+
+        const DynamicModelDefinition& modelDef = models_.at(automaton.second.id);
+        for (const auto &macroConnect : automaton.second.macroConnects) {
+            auto found = std::find_if(modelDef.nodeConnections.begin(), modelDef.nodeConnections.end(),
+            [&macroConnect](const algo::DynamicModelDefinition::MacroConnection &macroConnection) {
+                return macroConnection.id == macroConnect.macroConnection;
+            });
+            if (found == modelDef.nodeConnections.end()) {
+                if (automaton.second.lib == dfl::common::constants::svcModelName) {
+                    for (const auto &automatonGeneratorMacroConnect : automaton.second.macroConnects) {
+                        std::string genId = automatonGeneratorMacroConnect.id;
+                        if (assembling_.isSingleAssociation(genId)) {
+                        const auto &assoc = assembling_.getSingleAssociation(genId);
```

Comment:
as we have access to the dynamicMOdel.nodeConnections I don't think we need to use the assembling